lock-api/src/main/java/com/palantir/lock/v2/IdentifiedTimeLockRequest.java

```diff
@@ -24,15 +24,15 @@
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 
 @Value.Immutable
-@JsonSerialize(as = ImmutableLockImmutableTimestampRequest.class)
-@JsonDeserialize(as = ImmutableLockImmutableTimestampRequest.class)
-public interface LockImmutableTimestampRequest {
+@JsonSerialize(as = ImmutableIdentifiedTimeLockRequest.class)
+@JsonDeserialize(as = ImmutableIdentifiedTimeLockRequest.class)
+public interface IdentifiedTimeLockRequest {
```

Comment:
Why the rename? Will the old TimeLock be able to deserialize this as a `LockImmutableTimestampRequest`?

Reply:
I renamed this because calling an endpoint called `startAtlasDbTransaction` (for that matter, anything not named `lockImmutableTimestamp` or something close) with a `LockImmutableTimestampRequest` seemed awkward to me.

Deserialization: yep, that's a valid concern - it does deserialize. I've added a deserialization test as well.